src/Core/Models/Domain/Send.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Bit.Core.Abstractions;
+using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.View;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.Models.Domain
+{
+    public class Send : Domain
+    {
+        public string Id { get; set; }
+        public string AccessId { get; set; }
+        public string UserId { get; set; }
+        public SendType Type { get; set; }
+        public CipherString Name { get; set; }
+        public CipherString Notes { get; set; }
+        public SendFile File { get; set; }
+        public SendText Text { get; set; }
+        public CipherString Key { get; set; }
+        public int? MaxAccessCount { get; set; }
+        public int AccessCount { get; set; }
+        public DateTime RevisionDate { get; set; }
+        public DateTime? ExpirationDate { get; set; }
+        public DateTime DeletionDate { get; set; }
+        public string Password { get; set; }
+        public bool Disabled { get; set; }
+
+        public Send() { }
+
+        public Send(SendData data, bool alreadyEncrypted = false) : base()
+        {
+            BuildDomainModel(this, data, new HashSet<string>{
+                "Id",
+                "AccessId",
+                "UserId",
+                "Name",
+                "Notes",
+                "Key"
+            }, alreadyEncrypted, new HashSet<string> { "Id", "AccessId", "UserId" });
+
+            Type = data.Type;
+            MaxAccessCount = data.MaxAccessCount;
+            AccessCount = data.AccessCount;
+            Password = data.Password;
+            Disabled = data.Disabled;
+            RevisionDate = data.RevisionDate;
+            DeletionDate = data.DeletionDate;
+            ExpirationDate = data.ExpirationDate;
+
+            switch (Type)
+            {
+                case SendType.Text:
+                    Text = new SendText(data.Text, alreadyEncrypted);
+                    break;
+                case SendType.File:
+                    File = new SendFile(data.File, alreadyEncrypted);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public async Task<SendView> DecryptAsync()
+        {
+            var view = new SendView(this);
+
+            var cryptoService = ServiceContainer.Resolve<ICryptoService>("cryptoService");
+
+            try
+            {
+                view.Key = await cryptoService.DecryptToBytesAsync(Key, null);
+                view.CryptoKey = await cryptoService.MakeSendKeyAsync(view.Key);
+            }
+            catch (Exception e)
+            {
+                // TODO: error?
```

Comment:
@mportune-bw , any thoughts on whether this should be bubbled up? Not sure how logging or error handling is done in general for a scenario like this (coming out of a domain model); assuming we don't want an app crash, but at the same time behavior may be unexpected if the keys aren't set here.

Reply:
I would track the exception in AppCenter (via `Crashes.TrackError(e);`) then return null from this method so the caller can present "there was a problem" in the UI.

Reply:
Discussed with Matt. Since this is in `Core`, the right thing to do is just throw here and trust the `App` project to handle it.

In light of that, I'm just taking out the try/catch entirely.